javascript/webhooks-nodejs/createWebhook.js

```diff
@@ -0,0 +1,106 @@
+const axios = require("axios");
+const fs = require("node:fs");
+const path = require("node:path");
+require("dotenv").config();
+
+// File path for the .env file
+const envFilePath = path.join(__dirname, ".env");
+
+// Helper function to update the X-Hook-Secret in the .env file
+function updateXHookSecret(newSecret) {
+  let envContent = fs.readFileSync(envFilePath, "utf8");
+  envContent = envContent.replace(
+    /X_HOOK_SECRET=.*/,
+    `X_HOOK_SECRET=${newSecret}`
+  );
+  fs.writeFileSync(envFilePath, envContent);
+  console.log(`The X-Hook-Secret stored in .env is: ${newSecret}.`);
+}
+
+// Helper function to read X-Hook-Secret from the .env file
+function getXHookSecret() {
+  const envContent = fs.readFileSync(envFilePath, "utf8");
+  const match = envContent.match(/X_HOOK_SECRET=(.*)/);
+  return match ? match[1] : "";
+}
+
+// Async function to delete the webhook
+// Documentation: https://developers.asana.com/reference/deletewebhook
+async function deleteWebhook(webhookId, personalAccessToken) {
+  const url = `https://app.asana.com/api/1.0/webhooks/${webhookId}`;
+
+  try {
+    await axios.delete(url, {
+      headers: {
+        Authorization: `Bearer ${personalAccessToken}`,
+      },
+    });
+    console.log(`Webhook ${webhookId} deleted due to mismatch.`);
+  } catch (error) {
+    console.error(
+      "Error deleting webhook:",
+      error.response ? error.response.data : error.message
+    );
+  }
+}
+
+// Async function to create the webhook
+// Documentation: https://developers.asana.com/reference/createwebhook
+async function createWebhook(targetUri, objectId, filter, resourceType) {
+  const personalAccessToken = process.env.PAT;
+  const url = "https://app.asana.com/api/1.0/webhooks";
+
+  try {
+    const response = await axios.post(
+      url,
+      {
+        data: {
+          resource: objectId,
+          target: targetUri,
+          filters: [{ action: filter, resource_type: resourceType }],
+        },
+      },
+      {
+        headers: {
+          Authorization: `Bearer ${personalAccessToken}`,
+        },
+      }
+    );
+
+    const xHookSecret = response.data["X-Hook-Secret"];
+    const storedSecret = getXHookSecret();
+
+    // Compare the secrets. This check is not solely for detecting mutations in the stored secret; rather, it ensures that the original secret was valid and corresponds to what Asana expects
```

Comment:
**Nit:** Can you add new lines to this comment so we don't have to scroll horizontally to view
![scroll_horizontally](https://github.com/user-attachments/assets/173511f1-4fb4-4fe6-aaf6-5abb231512d7)

Reply:
Fixed!